File: leo2-ui/src/main/java/org/deku/leo2/bridge/services/MessageService.java
package org.deku.leo2.bridge.services;

import org.deku.leo2.bridge.IMessageService;
import org.deku.leo2.bridge.LeoBridge;
import org.deku.leo2.bridge.MediaType;
import org.deku.leo2.bridge.Message;

import javax.ws.rs.Consumes;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import java.util.Date;

/**
 * Created by masc on 17.09.14.
 */
@Path("/")
@Produces(MediaType.APPLICATION_JSON_UTF8)
@Consumes(MediaType.APPLICATION_JSON_UTF8)
public class MessageService implements IMessageService {
    public interface Listener {
        void onLeoBridgeServiceMessageReceived(Message message);
    }

    @Override
    public void send(Message message) {
        Date test = new Date();
        Listener l = (Listener) LeoBridge.instance();
        l.onLeoBridgeServiceMessageReceived(message);
    }
}

File: leoz-ui/src/main/java/org/deku/leoz/ui/bridge/services/MessageService.kt
package org.deku.leoz.ui.bridge.services

import com.github.salomonbrys.kodein.Kodein
import com.github.salomonbrys.kodein.conf.global
import com.github.salomonbrys.kodein.instance
import org.deku.leoz.ui.bridge.IMessageService
import org.deku.leoz.ui.bridge.LeoBridge
import org.deku.leoz.ui.bridge.MediaType
import org.deku.leoz.ui.bridge.Message
import javax.ws.rs.Consumes
import javax.ws.rs.Path
import javax.ws.rs.Produces

/**
 * Created by masc on 17.09.14.
 */
@Path("/")
@Produces(MediaType.APPLICATION_JSON_UTF8)
@Consumes(MediaType.APPLICATION_JSON_UTF8)
class MessageService : IMessageService {
    /** Leo bridge singgleton instance */
    val leoBridge: LeoBridge = Kodein.global.instance()

    interface Listener {
        fun onLeoBridgeServiceMessageReceived(message: Message)
    }

    override fun send(message: Message) {
        this.leoBridge.onLeoBridgeServiceMessageReceived(message)
    }
}
